algorithms/src/Kokkos_Random.hpp

```diff
@@ -1139,33 +1136,29 @@ class Random_XorShift1024_Pool {
  public:
   using generator_type = Random_XorShift1024<DeviceType>;
 
-  KOKKOS_INLINE_FUNCTION
-  Random_XorShift1024_Pool() { num_states_ = 0; }
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+  KOKKOS_FUNCTION
+  Random_XorShift1024_Pool() {
+    KOKKOS_IF_ON_DEVICE(
+        /* nonsensical initialization in the name of backward compatibility */
+        num_states_ = 0;)
+    KOKKOS_IF_ON_HOST(init(0, execution_space().concurrency());)
+  }
 
-  inline Random_XorShift1024_Pool(uint64_t seed) {
+  Random_XorShift1024_Pool(uint64_t seed) {
     num_states_ = 0;
 
     init(seed, execution_space().concurrency());
   }
+#else
+  Random_XorShift1024_Pool(uint64_t seed = 0) {
+    num_states_ = 0;
 
-  KOKKOS_INLINE_FUNCTION
-  Random_XorShift1024_Pool(const Random_XorShift1024_Pool& src)
-      : locks_(src.locks_),
-        state_(src.state_),
-        p_(src.p_),
-        num_states_(src.num_states_) {}
-
-  KOKKOS_INLINE_FUNCTION
-  Random_XorShift1024_Pool operator=(const Random_XorShift1024_Pool& src) {
-    locks_      = src.locks_;
-    state_      = src.state_;
-    p_          = src.p_;
-    num_states_ = src.num_states_;
-    padding_    = src.padding_;
-    return *this;
+    init(seed, execution_space().concurrency());
   }
+#endif
 
-  inline void init(uint64_t seed, int num_states) {
+  void init(uint64_t seed, int num_states) {
```

Comment:
same comment about this being a public method

Reply:
Marking it as revolved, will make sure that we are consistent.